lib/pipeline.js

```diff
@@ -1273,7 +1273,11 @@ class PipelineModel extends BaseModel {
                         collection.pipelineIds.includes(this.id));
 
                     return Promise.all(filteredCollections.map((collection) => {
-                        collection.pipelineIds.splice(collection.pipelineIds.indexOf(this.id), 1);
+                        const newPipelineIds = collection.pipelineIds.filter(pipelineId =>
```

Comment:
Nice finding!
https://stackoverflow.com/questions/37805412/remove-an-element-from-an-array-in-javascript-with-slice-or-spread